### PR TITLE
Reformat benchmark metrics

### PR DESCRIPTION
**Pull request by yeandy**
As part of effort the to automate benchmarks in XLML, the benchmark_serving.py script will be used to benchmark MaxText JetModels. The benchmark results, however, need to be in a slightly different format:
```
{
  "metrics": {"step_time": 100},
  "dimension": {"framework": "jax", ...}
}
```
This PR refactors the output, casts the values, and returns it for later use.

### CODE REVIEW

**Comment by yeandy** [comment]
@JoeZijunZhou Is the key [request_inthroughput](https://github.com/google/JetStream/pull/42/files#diff-214ef8ae13652260e1af7072a50af7a117523a4c57f918688309452c17994dacR474) a typo? should it be `request_throughput`?

**Comment by JoeZijunZhou** [comment]
> @JoeZijunZhou Is the key [request_inthroughput](https://github.com/google/JetStream/pull/42/files#diff-214ef8ae13652260e1af7072a50af7a117523a4c57f918688309452c17994dacR474) a typo? should it be `request_throughput`?

Yes, plz change it to `request_throughput`, thanks!

**Review by JoeZijunZhou — APPROVED** [review]
LGTM! Shall we also update the README in /benchmarks to give an example of running benchmark with eval? Also resolve the python checks before merge: https://github.com/google/JetStream/actions/runs/8742895378/job/23992198244?pr=42. Thanks!

File: benchmarks/benchmark_serving.py [hunk 1]
```diff
@@ -735,6 +748,12 @@ def main(args: argparse.Namespace):
       default="/tmp/request-outputs.json",
       help="File path to store request outputs",
   )
+  parser.add_argument(
+      "--run-eval",
```
Comment (FanhaiLu1) [inline benchmarks/benchmark_serving.py:752]: Can you add this change to readme?

Comment (FanhaiLu1) [reply benchmarks/benchmark_serving.py:752]: pylint check failed, could you fix the check error?

**Review by FanhaiLu1 — APPROVED** [review]

**Comment by vipannalla** [comment]
> LGTM! Shall we also update the README in /benchmarks to give an example of running benchmark with eval? Also resolve the python checks before merge: https://github.com/google/JetStream/actions/runs/8742895378/job/23992198244?pr=42. Thanks!
Agree, need to update our documentation.

File: benchmarks/benchmark_serving.py [hunk 2]
```diff
@@ -577,38 +578,50 @@ def main(args: argparse.Namespace):
...
+    metrics_json["request_rate"] = (
         args.request_rate if args.request_rate < float("inf") else "inf"
     )
 
-    # Merge with benchmark result
-    result_json = {**result_json, **benchmark_result}
+    metrics_json = {**metrics_json, **benchmark_result}
+    if args.run_eval:
+      eval_json = eval_accuracy(output)
+      metrics_json = {**metrics_json, **eval_json}
```
Comment (vipannalla) [inline benchmarks/benchmark_serving.py:606]: It maybe work keeping metrics separated into "perf" and "eval" categories. What is the interface to XLML storage -- are you flattening all metrics into one table?

**Review by vipannalla — COMMENTED** [review]
This is a much needed change. Makes it easy to run perf and eval in one command.

File: benchmarks/benchmark_serving.py (hunk 2 above)
Comment (yeandy) [reply benchmarks/benchmark_serving.py:606]: The `metrics` json is a single-level json. During json-lines [processing](https://github.com/GoogleCloudPlatform/ml-auto-solutions/blob/master/xlml/utils/metric.py#L181), it uses the assumption that the key is a string and value is a float (not another json).

It populates BigQuery tables in normalized tables. I'll show you what I mean via chat.

Also, see example in this test https://github.com/GoogleCloudPlatform/ml-auto-solutions/blob/master/xlml/utils/metric_test.py#L155

File: benchmarks/benchmark_serving.py (hunk 1 above)
Comment (yeandy) [reply benchmarks/benchmark_serving.py:755]: Done.

**Pull request merged**